assets/js/pages/UpgradablePackagesPage/UpgradablePackages.jsx

```diff
@@ -15,14 +17,67 @@ export default function UpgradablePackages({
   onLoad = noop,
 }) {
   const [search, setSearch] = useState('');
+  const enrichedPackages = upgradablePackages.map((packageDetails) => {
+    const { name, from_version, from_release, to_version, to_release, arch } =
+      packageDetails;
 
-  const displayedPackages = upgradablePackages.filter(
+    return {
+      ...packageDetails,
+      installed_package: `${name}-${from_version}-${from_release}.${arch}`,
+      latest_package: `${name}-${to_version}-${to_release}.${arch}`,
+    };
+  });
+  const displayedPackages = enrichedPackages.filter(
     ({ name, patches }) =>
       containsSubstring(name, search) ||
       patches
         .map(({ advisory }) => containsSubstring(advisory, search))
         .includes(true)
   );
+  const [csvURL, setCsvURL] = useState(null);
+
+  const hasPatches = (packages) =>
+    packages.some((packageDetails) => !isEmpty(packageDetails.patches));
+  const createCsvPackages = (packages) =>
```

Comment:
I'm not sure about this function.
You are replacing the initial `patches` by the advisories included in the patches.
Are we sure we want that? Would be a bad thing to send the complete data of the `patches`?

Either way, do we need a function for this?
Wouldn't be better to simply transform the data into a variable and use it in the `useEffect` (and you could change the variable used to trigger the useEffect as well